crates/secrets-types/src/lib.rs

```diff
@@ -187,6 +189,16 @@ impl SecretConfig {
 impl TryInto<HashMap<String, String>> for SecretConfig {
     type Error = anyhow::Error;
 
+    /// Convert this SecretConfig into a HashMap of the form:
+    /// ```json
+    /// {
+    ///   "type": "secret.wasmcloud.dev/v1alpha1",
+    ///   "backend": "baobun",
+    ///   "key": "/path/to/secret",
+    ///   "version": "vX.Y.Z",
+    ///   "policy": "{\"type\":\"properties.secret.wasmcloud.dev/v1alpha1\",\"properties\":{\"key\":\"value\"}}
```

Comment:
Based on the [test below](https://github.com/wasmCloud/wasmCloud/pull/2564/files#diff-140b7c9d1fe7f4ce11beb6d8d2bc2bfef1fc68bb3f5b9c3751f21d8066bd450bR37), should this be:
```suggestion
    ///   "policy": "{\"type\":\"policy.secrets.wasmcloud.dev/v1alpha1\",\"properties\":{\"key\":\"value\"}}
```

Reply:
That test is actually wrong 😄 it should be properties. Will fix

Reply:
For context, the `type` field within the `policy` is supposed to be the type+version of the properties schema. When / if we decide to update the structure of the `properties` block, we could update the type

Reply:
Makes sense, I appreciate the context!